lib/modules/datasource/go/index.ts

```diff
@@ -36,9 +36,8 @@ export class GoDatasource extends Datasource {
     key: ({ packageName }: Partial<DigestConfig>) => `${packageName}-digest`,
   })
   getReleases(config: GetReleasesConfig): Promise<ReleaseResult | null> {
-    return process.env.GOPROXY
-      ? this.goproxy.getReleases(config)
-      : this.direct.getReleases(config);
+    process.env.GOPROXY ??= 'https://proxy.golang.org,direct';
```

Comment:
I put it here so that users can configure this in `config.js` and take priority.

Reply:
shouldn't we also take customEnvVars into account?

Reply:
We don't today

Reply:
would prefer to move it to
https://github.com/renovatebot/renovate/blob/main/lib/modules/datasource/go/releases-goproxy.ts#L36

and don't change process ENV

🤔

Reply:
But it also needs to be passed to child process during artifacts updating later

Reply:
then we should update it there too. would prefer to have it closer to usage. otherwise we have a hidden side effect for all other renovate parts which can cause strange bugs later.

Reply:
I'm not sure it can be done elegantly as it's referenced in a few places